gitoxide-core/src/repository/log.rs

```diff
@@ -0,0 +1,152 @@
+use gix::bstr::{BStr, BString, ByteSlice};
+use gix::prelude::FindExt;
+use gix::ObjectId;
+
+pub fn log(mut repo: gix::Repository, out: &mut dyn std::io::Write, pathspec: BString) -> anyhow::Result<()> {
```

Comment:
`pathspec` definitely isn't a `pathspec`, and implementing it efficiently will be some work to get right, probably nothing you'd want to do here.
If it was just a path, it's more straightforward.

Reply:
I can’t seem to remember why I named it `pathspec`, but you’re right, at this point it’‘s definitely just a path.